library/std/src/sys/windows/path/absolute/tests.rs

```diff
@@ -0,0 +1,149 @@
+// Original implementation taken from crate "normpath".
+// Copyright 2020 "dylni" (https://github.com/dylni)
+
+// Most test cases were taken from:
+// https://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html
+
+// These tests assume that the current drive is "C:".
+
+use crate::env;
+use crate::io;
+use crate::path::Path;
+
+fn assert_eq(left: &Path, right: &Path) {
+    assert_eq!(left.as_os_str().bytes(), right.as_os_str().bytes());
+}
+
+fn test(expected: &str, path: &str) {
+    assert_eq(Path::new(expected), &super::absolute(Path::new(path)).unwrap());
+}
+
+macro_rules! test_unchanged {
+    ( $path:expr ) => {
+        test($path, $path);
+    };
+}
+
+#[test]
+fn test_drive_absolute() {
+    test_unchanged!(r"C:\ABC\DEF");
+    test_unchanged!(r"C:\");
+    test_unchanged!(r"C:\ABC\");
+    test(r"C:\ABC\DEF", r"C:\ABC\DEF. .");
+    test(r"C:\ABC\DEF", r"C:/ABC/DEF");
+    test(r"C:\XYZ", r"C:\ABC\..\XYZ");
+    test(r"C:\", r"C:\ABC\..\..\..");
+}
+
+#[test]
+fn test_drive_relative() {
+    test(r"X:\DEF\GHI", r"X:DEF\GHI");
+    test(r"X:\", r"X:");
+    test(r"X:\DEF", r"X:DEF. .");
+    test(r"Y:\", r"Y:");
+    test(r"Z:\", r"Z:");
+    test(r"X:\XYZ", r"X:ABC\..\XYZ");
+    test(r"X:\", r"X:ABC\..\..\..");
+}
+
+#[test]
+fn test_rooted() {
+    test(r"C:\ABC\DEF", r"\ABC\DEF");
+    test(r"C:\", r"\");
+    test(r"C:\ABC\DEF", r"\ABC\DEF. .");
+    test(r"C:\ABC\DEF", r"/ABC/DEF");
+    test(r"C:\XYZ", r"\ABC\..\XYZ");
+    test(r"C:\", r"\ABC\..\..\..");
+}
+
+#[test]
+fn test_unc_absolute() {
+    test_unchanged!(r"\\server\share\ABC\DEF");
+    test_unchanged!(r"\\server\share");
+    test(r"\\server\share\ABC", r"\\server\share\ABC. .");
+    test(r"\\server\share\ABC\DEF", r"//server/share/ABC/DEF");
+    test(r"\\server\share\XYZ", r"\\server\share\ABC\..\XYZ");
+    test(r"\\server\share", r"\\server\share\ABC\..\..\..");
+
+    assert_eq!(
+        "partial UNC prefixes are invalid",
+        super::absolute(Path::new(r"\\server")).unwrap_err().to_string(),
+    );
+}
+
+#[test]
+fn test_local_device() {
+    test_unchanged!(r"\\.\COM20");
+    test_unchanged!(r"\\.\pipe\mypipe");
+    test(r"\\.\C:\ABC\DEF", r"\\.\C:\ABC\DEF. .");
+    test(r"\\.\C:\ABC\DEF", r"\\.\C:/ABC/DEF");
+    test(r"\\.\C:\XYZ", r"\\.\C:\ABC\..\XYZ");
+    test(r"\\.\X:\", r"\\.\C:\ABC\..\..\X:\");
+    test(r"\\.\pipe\notmine", r"\\.\pipe\mypipe\..\notmine");
+
+    test(r"\\.\COM1", r"COM1");
+    test(r"\\.\COM1", r"C:\COM1");
+    test(r"\\.\COM1", r"C:COM1");
+    test(r"\\.\COM1", r"valid\COM1");
+    test(r"\\.\COM1", r"C:\notvalid\COM1");
+    test(r"\\.\COM1", r"C:\COM1.blah");
+    test(r"\\.\COM1", r"C:\COM1:blah");
+    test(r"\\.\COM1", r"C:\COM1  .blah");
```

Comment:
Windows 11 has changed these. They will no longer resolve to `\\.\COM1` but instead are treated as regular files. See also: https://github.com/rust-lang/cargo/pull/10322

Reply:
That's surprising. Do you have a reference? I'd be interested to read more about the change.

Reply:
Unfortunately the change is currently undocumented but it can be shown by running tests on Windows 11. Essentially it now only matches the exact (case-insensitive) name. So `cOm1` still resolves to `\\.\COM1` but `COM1.txt` or `C:\COM1` are treated normally.

Reply:
Still, that's good to hear. I find it strange that Windows would break backward compatibility for such an old feature and not mention it, but it might have such little use that it's not an issue.